### PR TITLE
Use source model EQLNUM region when generating a FlowNet

### LINKED ISSUE #139: High degrees of freedom with individual satnum/eqlnum
When simulating Norne, there are clearly different equilibration regions. Also tuning the relative permeability individually will be beneficial for a well-by-well match. However, when options are set to individual, without adding a single additional node, the degrees of freedom end up at nearly 30,000, which is about 4 times the number of 'observations'.

We should consider limiting the number of equilibration/relative permeability regions to have the right amount of flexibility. A simple first step could be to take the original simulation regions and assign the same regions number in the same locations.

This would in essence not violate the data-driven principle of FlowNet as we could assign regions based on well properties. But before answering that question, testing the principle based on simulation model output could be useful.

### PR DESCRIPTION
**Pull request by olelod**
There are a lot of parameters to history match when we use individual SATNUM/EQLNUM. To limit this number we can use the original models' SATNUM/EQLNUM.

---

### Contributor checklist

- [ ] :tada: This PR closes #139.
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Assign eqlnum regions from simulation to FlowNet model.
   - [ ] Assign satnum regions from simulation to FlowNet model
   - [x] Testing
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`.
- [ ] :books: I have considered updating the documentation.

### CODE REVIEW

File: src/flownet/ahm/_run_ahm.py [hunk 1]
```diff
@@ -22,6 +23,69 @@
...
+        name: The same of the region parameter
+
+    Returns:
+        A list with values for 'name' region for each tube in the FlowNet model
+    """
+    df_regions = []
+
+    x_mid = network.grid[["x0", "x1", "x2", "x3", "x4", "x5", "x6", "x7"]].mean(axis=1)
+    y_mid = network.grid[["y0", "y1", "y2", "y3", "y4", "y5", "y6", "y7"]].mean(axis=1)
+    z_mid = network.grid[["z0", "z1", "z2", "z3", "z4", "z5", "z6", "z7"]].mean(axis=1)
```
Comment (wouterjdb) [inline src/flownet/ahm/_run_ahm.py:52]: Maybe make a property out of this.

You can copy that from my PR:

https://github.com/equinor/flownet/pull/155/files

**Review by wouterjdb — CHANGES_REQUESTED** [review]

File: src/flownet/data/from_flow.py [hunk 2]
```diff
@@ -324,6 +325,14 @@ def _grid_cell_bounding_boxes(self) -> np.ndarray:
     def _get_start_date(self):
         return self._eclsum.start_date
 
+    def init(self, name: str) -> np.ndarray:
+        """array with 'name' regions"""
+        return self._init[name][0]
+
+    def get_unique_regions(self, name) -> np.ndarray:
```
Comment (wouterjdb) [inline src/flownet/data/from_flow.py:332]:
```suggestion
    def get_unique_regions(self, name: str) -> np.ndarray:
```

**Review by wouterjdb — CHANGES_REQUESTED** [review]

File: src/flownet/ahm/_run_ahm.py [hunk 3]
```diff
@@ -22,6 +23,62 @@
...
 
 
+def _from_regions_to_flow_tubes(
+    network: NetworkModel,
+    field_data: FlowData,
+    ti2ci: pd.DataFrame,
+    name: str,
+) -> List[int]:
+    """
+        The function loops through each cell in all flow tubes, and checks what 'name' region the
```
[the remark this answers is not in the document]
Comment (olelod) [reply src/flownet/ahm/_run_ahm.py:33]: Name is an input: EQLNUM or SATNUM

[the remark this answers is not in the document]
Comment (olelod) [reply src/flownet/ahm/_run_ahm.py]: Slightly rewritten in new commit

**Review by wouterjdb — APPROVED** [review]
LGTM 🚀

**Pull request merged**